src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -1090,3 +1090,27 @@ def forward(
         return CausalLMOutput(
             loss=loss, logits=logits, hidden_states=outputs.hidden_states, attentions=outputs.attentions
         )
+
+    def resize_lm_head(self, new_num_tokens: Optional[int] = None) -> torch.nn.Linear:
```

Comment:
Could we remove this change from the PR, it's better to focus this PR on the examples only. Also, I'm not very much in favor of providing a `resize_lm_head()` method to the model yet since different to *e.g.* BERT a speech model has no word embedding layer during pertaining, it is only when the model is fine-tuned that a lm head is needed. So if one wants to resize the lm head that means that the model has to be fine-tuned at least twice, which is quite an edge case at the moment IMO. 

I'd prefer to not add this for now

Reply:
Yep, can you suggest a different way to increase the size of the linear layer's output to match bigger vocabs (e.g., for Arabic, it's 56)? I can move this method to `run_asr.py` or follow another approach.
I don't think it's about fine-tuning twice; it's about increasing the size of this layer's output:
https://github.com/huggingface/transformers/blob/c83fbc5f2d4355a13be52b086fe548f3c43e8ec8/src/transformers/models/wav2vec2/modeling_wav2vec2.py#L978

Reply:
Gottcha, yeah in this case, I don't think we need any resizing method at all! What I'd suggest is to do the following. 
1. First instantiate the tokenizer with the vocab as you wish. The correct vocab size is then defined by ```len(tokenizer)``.
2. Second, load the model. Now you can load the model as follows: ```model = Wav2Vec2ForCTC.from_pretrained("facebook/wav2vec2-base", vocab_size=len(tokenizer))```. The model should then have the correct lm head size. 

Also note that all "pretrained-only" checkpoints don't have any weights for the `lm_head` you can see this when *e.g.* loading the model: ```Wav2Vec2ForCTC.from_pretrained("facebook/wav2vec2-base")``` should show a warning that says that the lm head was randomly initialized because those weights aren't stored in the checkpoint. I tried to explain this in the blog as well: https://huggingface.co/blog/fine-tune-xlsr-wav2vec2

Reply:
Yeah I'm aware of this being a new layer (model head) added in model class; I need to get more familiar with the various kwargs and docs — I should have asked you long time ago instead of tracing the code :) Thanks!